src/include/index/ivf_pq_index.h

```diff
@@ -813,6 +839,37 @@ class ivf_pq_index {
     return pq_vectors;
   }
 
+  template <
+      feature_vector_array U,
+      class Matrix,
+      class Distance = uncached_sub_sum_of_squares_distance>
+  auto pq_query_centroid_distances(
+      const U& training_set, Distance distance = Distance{}) const {
```

Comment:
(Minor)(Optional) Might be a bit easier to read if we just refer to these as `vectors`:
```suggestion
      const U& vectors, Distance distance = Distance{}) const {
```

Reply:
Renamed to `query_vectors`